packages/devtools/lib/src/memory/memory.dart

```diff
@@ -362,38 +682,357 @@ class MemoryScreen extends Screen with SetStateMixin {
-        tableContainer.add(_createInstanceView(
-          instance != null
-              ? row.objectRef
-              : 'Unable to fetch instance ${row.objectRef}',
-          row.className,
-        ));
-
-        tableContainer.element.scrollTo(<String, dynamic>{
-          'left': tableContainer.element.scrollWidth,
-          'top': 0,
-          'behavior': 'smooth',
-        });
+  void select(InstanceSummary row) async {
+    ga.select(ga.memory, ga.inspectInstance);
+
+    // User selected a new instance from the list of class instances so the
+    // instance view which would be the third child needs to be removed.
+    _removeInstanceView();
+
+    Instance instance;
+    try {
+      instance = await memoryController.getObject(row.objectRef);
+
+      final dynamic theObject = await memoryController.getObject(row.objectRef);
+      if (theObject is Instance) {
+        instance = theObject;
+      } else if (theObject is Sentinel) {
+        instance = null;
+        // TODO(terry): Tracking Sentinel's to be removed.
+        framework.toast('Sentinel ${row.objectRef}', title: 'Warning');
+      }
+    } catch (e) {
+      instance = null; // Signal a problem
+    } finally {
+      tableContainer.add(_createInstanceView(
+        instance != null
+            ? row.objectRef
+            : 'Unable to fetch instance ${row.objectRef}',
+        row.className,
+      ));
+
+      tableContainer.element.scrollTo(<String, dynamic>{
+        'left': tableContainer.element.scrollWidth,
+        'top': 0,
+        'behavior': 'smooth',
+      });
+
+      // Allow inspection of the memory object.
+      memoryDataView.showFields(instance != null ? instance.fields : []);
+
+      // Record this navigation.
+      // TODO(terry): Is there something faster for identity compare?
+      final InstanceRef eval= await evaluate(row.objectRef, 'hashCode');
+      final int evalResult = int.parse(eval?.valueAsString);
+
+      if (!fromMemoryHover &&
+          (memoryPath.isLastInBound || memoryPath.isLastInstance)) {
+        // User clicked an instance, start new history with this instance.
+        _resetHistory();
+      }
+
+      // Record the memory navigation.
+      memoryPath.add(NavigationState.instanceSelect(row.className, evalResult));
+
+      if (memoryPath.isLastInBound) {
+        // Re-construct memory navigation and display.
+        history.clear();
+        memoryPath.addPathsLinks(history, _handleHistoryClicks);
+
+        history..clazz('history-navigation', removeOthers: true);
+      }
+    }
+  }
+
+  void _handleHistoryClicks(CoreElement element) {
+    // Handle clicking in the history links.
+    if (element.tag == 'SPAN' && element.hasClass('history-link')) {
+      final attrs = element.attributes;
+
+      final int dataIndex = int.parse(attrs[NavigationState.dataIndex]);
+
+      final String dataClass = attrs[NavigationState.dataClass];
+      final String dataField = attrs.containsKey(NavigationState.dataField)
```

Comment:
nit: can just be 
attrs[NavigationState.dataField] ?? ''

Reply:
Could be null if specified w/o a value, changed to:

      String dataField = attrs[NavigationState.dataField];
      dataField ??= '';

more efficient.